fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/TransactionImpl.java

```diff
@@ -30,40 +37,69 @@
  */
 public class TransactionImpl implements Transaction {
 
+    private final static Logger log = LoggerFactory.getLogger(TransactionImpl.class);
+
     final String id;
 
+    final TransactionManagerImpl txManager;
+
+    final Duration DEFAULT_TIMEOUT = ofMinutes(3);
+
     boolean shortLived = true;
 
-    TransactionImpl(final String id) {
+    Instant expiration;
+
+    boolean expired = false;
+
+    boolean rolledback = false;
+
+    boolean commited = false;
+
+    protected TransactionImpl(final String id, final TransactionManagerImpl txManager) {
         if (id == null || id.isEmpty()) {
             throw new IllegalArgumentException("Transaction id should not be empty!");
         }
         this.id = id;
+        this.txManager = txManager;
+        this.expiration = Instant.now().plus(timeout());
     }
 
     @Override
     public void commit() {
```

Comment:
Transaction instances will be shared across requests, won't they? Se we probably need some kind of locking or synchronization scheme so that we don't run into race conditions with simultaneous commit/rollback/expire/update calls. I'm not sure if we need anything more sophisticated than making most of the public methods synchronized. The persistent session classes are taking care of a lot of the more exact locking